### ISSUE TITLE
Add "-rate" option to simulate overload

### ISSUE THREAD

**Issue by nyh** [scylladb/scylla-bench#42]
Scylla-bench has a "-max-rate" option which slows the client down if achieving more than the desired request rate.

To simulate **overload** situations with **unbounded client concurrency**, I want a different option - we can call it "-min-rate" (or something else less confusing) - which will try to achieve **at least** the desired rate. This desired rate will quite possibly be higher than the request rate that the server can successfully answer (that's the whole point of the overload simulation), so scylla-bench willl start to accumulate a lot of unanswered requests, and need to make sure that scylla-bench can send as many as alllowed unanswered requests on the same connection, and open new connections if necessary. Scylla-bench just needs to make sure that it *sends* the desired number of requests per second - regardless of how many answers come back, or what kind of answers these are (success, failure, timeout, etc.).

When "-min-rate" is used, the "-connection-count" option maybe should be ignored, or, maybe they should serve as upper limit. I don't know what "-concurrency" means in scylla-bench, if it is a limit on the number of unanswered request, it shouldn't exist at all in this mode.

CC @eliransin @psarna 

Migrated to Jira: [QATOOLS-51](https://scylladb.atlassian.net/browse/QATOOLS-51)

[QATOOLS-51]: https://scylladb.atlassian.net/browse/QATOOLS-51?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by psarna** [comment]
Good idea. I guess we can just call it `-rate`, since it shouldn't be that hard to simply send requests at exactly given rate, regardless of how many replies were received so far. As for the conflicting parameters, it would make sense to simply make some options mutually exclusive - e.g. if `-min-rate` (or just `-rate`) is set, it's an error to also set `-concurrency`. It also already happens for different parameters, e.g. you cannot pick a workload type if you already defined a `scan` mode.

**Comment by psarna** [comment]
@nyh and as for your `-concurrency` question, it's simply a number of fibers (goroutines) spawned to serve a workload. So indirectly it is indeed a limit of unanswered requests, since each fiber sends a request and then waits for an answer.

**Comment by nyh** [comment]
> @nyh and as for your `-concurrency` question, it's simply a number of fibers (goroutines) spawned to serve a workload. So indirectly it is indeed a limit of unanswered requests, since each fiber sends a request and then waits for an answer.

I see. So "concurrency" is similar to Seastar having a limit on the number of unresolved futures? Why should we have any such limit (except perhaps a ridiculously high limit based on the amount of memory)?

**Comment by dani-tweig** [comment]
Closing this issue as it was moved to Jira.
Please continue the thread in https://scylladb.atlassian.net/browse/QATOOLS-51
